src/test/java/com/fittracker/fittracker/controller/BaseIntegrationTest.java

```diff
@@ -50,4 +79,38 @@ void givenRequestWithNoAuthorization_shouldReturnUnauthorizedError() throws Exce
         }
     }
 
+    protected String makeRequest(String endpoint, HttpMethod httpMethod, HttpStatus expectedStatus) throws Exception {
+        return performRequest(endpoint, httpMethod, expectedStatus);
+    }
+
+    protected <T> T makeRequest(String endpoint, HttpMethod httpMethod, HttpStatus expectedStatus,
+        Class<T> responseClass) throws Exception {
+        var responseString = performRequest(endpoint, httpMethod, expectedStatus);
+
+        return mapper.readValue(responseString, responseClass);
+    }
+
+    protected <T> T makeRequestWithBody(HttpMethod httpMethod, Object requestBody, HttpStatus expectedStatus,
+        Class<T> responseClass) throws Exception {
+        return makeRequestWithBody(getEndpoint(), requestBody, httpMethod, expectedStatus, responseClass);
+    }
+
+    protected <T> T makeRequestWithBody(String endpoint, Object requestBody, HttpMethod httpMethod,
+        HttpStatus expectedStatus, Class<T> responseClass) throws Exception {
+        var responseString = mockMvc.perform(request(httpMethod, create(endpoint))
+                .header(AUTHORIZATION_HEADER_NAME, AUTHORIZATION_HEADER_VALUE_PREFIX + TOKEN)
+                .contentType(APPLICATION_JSON)
+                .content(mapper.writeValueAsString(requestBody)))
+            .andExpect(status().is(expectedStatus.value()))
+            .andReturn().getResponse().getContentAsString();
+
+        return mapper.readValue(responseString, responseClass);
+    }
```

Comment:
As the first method is only calling `getEndpoint()` and passing arguments to the second, lets get rid of first method, reorder arguments of the second one (so that it matches original order of first method) and lastly, substitute `String endpoint` with `getEndpoint()` call.

Something like this:
```
protected <T> T makeRequestWithBody(HttpMethod httpMethod, Object requestBody, HttpStatus expectedStatus,
         Class<T> responseClass) throws Exception {
        var responseString = mockMvc.perform(request(httpMethod, create(getEndpoint()))
                .header(AUTHORIZATION_HEADER_NAME, AUTHORIZATION_HEADER_VALUE_PREFIX + TOKEN)
                .contentType(APPLICATION_JSON)
                .content(mapper.writeValueAsString(requestBody)))
            .andExpect(status().is(expectedStatus.value()))
            .andReturn().getResponse().getContentAsString();

        return mapper.readValue(responseString, responseClass);
 }
```

Reply:
good catch, fixed